geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3BlobStoreInfo.java

```diff
@@ -334,7 +337,38 @@ public BlobStore createInstance(TileLayerDispatcher layers, LockProvider lockPro
         checkState(
                 isEnabled(),
                 "Can't call S3BlobStoreConfig.createInstance() is blob store is not enabled");
-        return new S3BlobStore(this, layers, lockProvider);
+        final GeoWebCacheEnvironment gwcEnvironment =
+                GeoWebCacheExtensions.bean(GeoWebCacheEnvironment.class);
+        return new S3BlobStore(this.clone(gwcEnvironment, true), layers, lockProvider);
+    }
+
+    public S3BlobStoreInfo clone(
+            GeoWebCacheEnvironment gwcEnvironment, Boolean allowEnvParametrization) {
+        S3BlobStoreInfo blobStore = SerializationUtils.clone(this);
+
+        if (allowEnvParametrization && gwcEnvironment != null) {
+            blobStore.setName(getName());
+            blobStore.setEnabled(isEnabled());
+            blobStore.setDefault(isDefault());
+            blobStore.setAccess(getAccess());
+            blobStore.setPrefix(getPrefix());
+            blobStore.setUseHTTPS(isUseHTTPS());
+            blobStore.setUseGzip(isUseGzip());
+
+            blobStore.setBucket((String) gwcEnvironment.resolveValue(getBucket()));
+            blobStore.setAwsAccessKey((String) gwcEnvironment.resolveValue(getAwsAccessKey()));
+            blobStore.setAwsSecretKey((String) gwcEnvironment.resolveValue(getAwsSecretKey()));
+            blobStore.setMaxConnections((Integer) gwcEnvironment.resolveValue(getMaxConnections()));
```

Comment:
I thought you reverted back to parameterizing only strings, but this is an Integer instead.

Reply:
ahh, missed it, have updated code